### PR TITLE
Make self-update optional, but still default

### LINKED ISSUE #202: Make self_update optional
`self_update` is quite useless on package managed systems (e.g., the AUR package). Would be cool if package builders could disable this feature.

### PR DESCRIPTION
**Pull request by rubdos**
Packagers can now use `--no-default-features` in order to disable self updates.
Closes #202

How do we test this in CI?

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -83,7 +86,9 @@ version = "0.5.1"
 fs_extra = "1.0.0"
 
 [features]
-beta = []
-default = []
+jeta = []
```
Comment (vitiral) [inline Cargo.toml]: did you mean to change this to "jeta"?

File: Cargo.toml [hunk 2]
```diff
@@ -83,7 +86,9 @@ version = "0.5.1"
...
+default = ["self-update"]
+self-update = ["self_update"]
```
Comment (vitiral) [inline Cargo.toml]: I think just make it

```
default = ["self_update"]
self_update = []
```

Then use `self_update` in all the cfg attributes.

Let me know if I'm wrong about this (I totally could be!)
```

**Comment by vitiral** [comment]
sorry that I forgot to get to this.

File: Cargo.toml (hunk 2 above)
Comment (rubdos) [reply Cargo.toml]:
```
error: failed to parse manifest at `/home/rsmet/src/artifact/Cargo.toml`

Caused by:
  Features and dependencies cannot have the same name: `self_update`
```

It seems to work without the `self_update = []` though.

File: Cargo.toml (hunk 1 above)
Comment (rubdos) [reply Cargo.toml]: Nowp, slip of the finger. Sorry.

**Comment by rubdos** [comment]
No problem, it's not urgent to me. Just wanted to help out here :-)

**Comment by vitiral** [comment]
Okay, so I think the issue is that "feature" and "dependency" can't have the same name. Can you change the feature to just `update` and have `update = []` in features.

Sorry again this took me a while. I didn't realize you had committed new changes.

**Comment by rubdos** [comment]
Ha, didn't notice the build failed, it worked on my side. Weird.

**Comment by rubdos** [comment]
Note that the build succeeded in travis, but it gave a Python error? Anyhow, I removed the last warning travis gave; apparently I didn't run a linter 😞

**Comment by vitiral** [comment]
I am doing a practically complete rewrite of artifact's cli and will implement this there.

Sorry for invalidating your work :(

**Comment by rubdos** [comment]
> I am doing a practically complete rewrite of artifact's cli and will implement this there.
> 
> Sorry for invalidating your work :(

No problem, thanks :-)
Give me a ping on gitter/IRC/here if it's there, then I can incorporate it into the Arch PKGBUILD.

**Pull request closed without merging**